File: recipesv2/src/recipe/tmp.py
from src.schema.enumeration import Repo, Scope
from src.schema.atomic import Package, Topic, Unit


map = {}

Topic(
    name="misc",
    recipes=[
        Package("espanso-wayland", Repo.UNKNOWN),
        Package("zenity", Repo.OFFICIAL),
        Package("papirus-icon-theme", Repo.OFFICIAL),
        Package("glance-bin", Repo.AUR),
        Package("gh-dash-bin", Repo.AUR),
        Package("witr-bin", Repo.AUR),
        Package("zerobrew-bin", Repo.AUR),

        Package("ast-grep", Repo.AUR),
        Package("uutils-coreutils", Repo.AUR),
        Package("pake", Repo.AUR),
        Package("sing-box-bin", Repo.AUR),
        Package("qbittorrent", Repo.OFFICIAL)
    ]
)

Topic(
    name="time",
    recipes=[
        Package("task", Repo.OFFICIAL),
        Package("timew", Repo.OFFICIAL),

        Package("watson", Repo.AUR),
        Package("activitywatch-bin", Repo.AUR),
        Package("hamster-time-tracker", Repo.OFFICIAL),
    ]
)

Topic(
    name="email",
    recipes=[
        Package("thunderbird", Repo.OFFICIAL),
        Package("betterbird-bin", Repo.AUR),
        Package("stalwart-mail", Repo.AUR),
        Package("stalwart-cli", Repo.AUR),
    ]
)

Topic(
    name="ai",
    recipes=[
        Package("gemini-cli", Repo.OFFICIAL),
        Package("openai-codex", Repo.OFFICIAL),

        Package("jan-bin", Repo.AUR),
        Package("cc-switch-bin", Repo.AUR),
        Package("chatgpt-desktop-bin", Repo.AUR),
        Package("nextchat-bin", Repo.AUR),
        Package("cherry-studio-bin", Repo.AUR),
        Package("gpt4all-chat-git", Repo.AUR),
        Package("anythingllm-desktop-bin", Repo.AUR),
        Package("open-webui", Repo.AUR),
        Package("ollama-bin", Repo.AUR),
        Package("goose-desktop-bin", Repo.AUR),
    ]
)

Topic(
    name="secret",
    recipes=[
        Package(
            name="seahorse",
            repo=Repo.OFFICIAL,
            supporters=[
                Package("gnome-keyring", Repo.OFFICIAL, units=[Unit("gnome-keyring-daemon.service", scope=Scope.SINGLE_USER)])
            ]
        ),
        Package(
            name="bitwardern",
            repo=Repo.OFFICIAL,
            supporters=[
                Package("gnome-keyring", Repo.OFFICIAL, units=[Unit("gnome-keyring-daemon.service", scope=Scope.SINGLE_USER)])]
        ),
    ]
)

Topic(
    name="backup",
    recipes=[
        Package("borg", Repo.OFFICIAL),
        Package("restic", Repo.OFFICIAL),
        Package("timeshift", Repo.OFFICIAL),
    ]
)

Topic(name="disk-remote-mount", recipes=[Package("sshfs", Repo.OFFICIAL)])

Topic(
    name="audio-gui",
    recipes=[
        Package("pavucontrol", Repo.OFFICIAL),
        Package("easyeffects", Repo.OFFICIAL),
        Package("qpwgraph", Repo.OFFICIAL),
    ]
)

Topic(
    name="firewall",
    recipes=[Package("ufw", Repo.OFFICIAL, supporters=[Package("gufw", Repo.OFFICIAL)]) ]
)

Topic(
    name="git-server",
    recipes=[
        Package("gitea", Repo.OFFICIAL),
        Package("forgejo", Repo.OFFICIAL)
    ]
)

Topic(
    name="office",
    recipes=[
        Package("onlyoffice", Repo.AUR),
        Package("libreoffice-fresh", Repo.OFFICIAL),
        Package("openoffice", Repo.OFFICIAL),
    ]
)

Topic(
    name="note-taking",
    recipes=[
        # Package("memos", Repo.AUR),
        # Package("zettlr ", Repo.AUR),
        Package("obsidian", Repo.OFFICIAL),

        # Package("appflowy-bin", Repo.AUR),
        # Package("logseq-desktop-bin", Repo.AUR),
        # Package("joplin-desktop", Repo.AUR),
        # Package("notesnook-bin", Repo.AUR),
        # Package("vnote-bin", Repo.AUR),
    ]
)

Topic(
    name="social",
    recipes=[
        # telegram
        Package("telegram-desktop", Repo.OFFICIAL),

        # discord
        Package("vesktop-bin", Repo.AUR),

        # matrix
        Package("fractal", Repo.OFFICIAL),
        Package("fluffychat-bin", Repo.AUR),

        # spotify
        Package(
            name="spotify",
            repo=Repo.AUR,
            single_user_config=["~/.config/spotify"],
            supporters=[Package("spicetify-cli", Repo.AUR, single_user_config=["~/.config/spicetify"])]
        ),
        Package("spotify-tui", Repo.OFFICIAL),
    ]
)

File: recipes/src/package/tmp.lua
dofile("utils.lua")

local m={}

m.misc={
    espanso_wayland={Repo.UNKNOWN},
    dotter={Repo.GITHUB},
    markdown_oxide=aor,
    zenity=aor,
    papirus_icon_theme=aor,
    glance_bin=aur,

    github_cli=aor,
    gh_dash_bin=aur,

    witr_bin=aur,
    zerobrew_bin=aur,
    stremio=aur;

    ast_grep=aur,
    uutils_coreutils=aur,
    pake=aur,
    sing_box_bin=aur,
    asciidoctor=aor,

    mise=aor,
    just=aor,
    direnv=aor,
    entr=aor,
    watchexec=aor,
    bear=aor,
    ctags=aor,
    compiledb=aur,
    freetube_bin=aur,
}

m.download={
    aria2=aor,
    -- yt_dlp=aor,
    -- qbittorrent=aur,
    -- freedownloadmanager=aur,
}

m.image.editor={
    gimp=aor,
    krita=aor,
    darktable=aor,
    pinta=aur,
    rawtherapee=aor,
    lazpaint_bin=aur,
}

m.pdf={
    zathura=aor,
    zathura_pdf_mupdf=aor,
    zathura_pdf_poppler=aor,
    evince=aor,
    okular=aor,
    mupdf=aor
}

m.archive={
    xarchiver=aor,
    file_roller=aor,
    -- engrampa=aor,
    backend={
        unzip=aor,
        seven_zip=aor,
        unrar=aor,
        tar=aor
    }
}

m.font_viewer={
    gnome_font_viewer=aor;
}

-- m.browser.keyboard_driven={
--     browsh=aur,
--     nyxt=aor,
--     lynx=aor,
--     luakit=aor,
-- }

m.time={
    task=aor,
    timew=aor,
    waston=aur,

    activitywatch_bin=aur,
    hamster_time_tracker=aur,
}

m.email={
    thunderbird=aor,
    betterbird_bin=aur,
    stalwart_cli=aur,
    stalwart_mail=aur,
}

m.ai={
    gemini_cli=aor,
    openai_codex=aor,

    jan_bin=aur,
    cc_switch_bin=aur,
    chatgpt_desktop_bin=aur,
    nextchat_bin=aur,
}

m.secret={
    seahorse={
        Repo.AOR,
        supporters={
            gnome_keyring={Repo.AOR, units={"gnome-keyring-daemon.service", scope=Scope.SINGLE_USER}}
        }
    },
    bitwardern={
        Repo.AOR,
        supporters={
            gnome_keyring={Repo.AOR, units={"gnome-keyring-daemon.service", scope=Scope.SINGLE_USER}}
        }
    }
}

m.backup={
    borg=aor,
    restic=aor,
    timeshift=aor,
}

m.disk.mount.remote.sshfs=aor

m.audio.gui={
    pavucontrol=aor,
    easyeffects=aor,
    qpwgraph=aor,
}

m.network.firewall.ufw={Repo.AOR, supporters={gufw=aor}}

m.git_server={
    gitea=aor,
    forgejo=aor,
}

m.office={
    onlyoffice=aur,
    openoffice=aur,
    libreoffice_fresh=aor,

    abiword=aor,
    gnumeric=aor,
}

m.social={
    telegram_desktop=aor,
    vesktop_bin=aur,
    fractal=aor,
    fluffychat_bin=aur,
    spotify={
        Repo.AUR,
        single_user_config={"~/.config/spotify"},
        supporters={spicetify_cli={Repo.AUR, single_user_config={"~/.config/spicetify"}}}
    },
    spotify_tui=aor,
}

m.note_taking={
    memos=aur,
    zettlr=aur,
    obsidian=aor,

    clickup=aur,
    appflowy_bin=aur,
    logseq_desktop_bin=aur,
    joplin_desktop=aur,
    notesnook_bin=aur,
    vnote_bin=aur,
}
